core/domain/collection_domain.py

```diff
@@ -907,132 +912,71 @@ def exploration_ids(self):
+            if exp_id not in completed_exp_ids:
+                return [exp_id]
+        return []
 
     def get_next_exploration_ids_in_sequence(self, current_exploration_id):
-        """Returns a list of exploration IDs that a logged-out user should
-        complete next based on the prerequisite skill ids they must have
-        attained by the time they completed the current exploration. This
-        recursively compiles a list of 'learned skills' then, depending on
-        the 'learned skills' and the current exploration's acquired skill ids,
-        returns either a list of exploration ids that have either just
-        unlocked or the user is qualified to explore.  If neither of these
-        lists can be generated a blank list is returned instead.
+        """Returns the exploration ID of the node just after the node
+           corresponding to the current exploration id. If the user is on the
+           last node, empty list is returned.
 
         Args:
             current_exploration_id: str. The id of exploration currently
                 completed.
 
         Returns:
-            list(str). List of exploration IDs that a logged-out user should
-            complete next.
+            list(str). A single element list of the exploration ID that a
+            logged-out user should complete next.
         """
-        skills_learned_by_exp_id = {}
-
-        def _recursively_find_learned_skills(node):
-            """Given a node, returns the skills that the user must have
-            acquired by the time they've completed it.
-
-            Arg:
-                node: CollectionNode. A node in the exploration graph of a
-                    collection.
-
-            Returns:
-                list(str). A list of skill ids acquired by user.
-            """
-            if node.exploration_id in skills_learned_by_exp_id:
-                return skills_learned_by_exp_id[node.exploration_id]
-
-            skills_learned = set(node.acquired_skill_ids)
-            for other_node in self.nodes:
-                if other_node.exploration_id not in skills_learned_by_exp_id:
-                    for skill in node.prerequisite_skill_ids:
-                        if skill in other_node.acquired_skill_ids:
-                            skills_learned = skills_learned.union(
-                                _recursively_find_learned_skills(other_node))
-
-            skills_learned_by_exp_id[node.exploration_id] = skills_learned
-            return skills_learned
+        exploration_just_unlocked = []
 
-        explorations_just_unlocked = []
-        explorations_qualified_for = []
-
-        collection_node = self.get_node(current_exploration_id)
-        collected_skills = _recursively_find_learned_skills(collection_node)
+        for index in range(0, len(self.nodes)-1):
+            if self.nodes[index].exploration_id == current_exploration_id:
+                exploration_just_unlocked = self.nodes[index+1].exploration_id
```

Comment:
This is confusing -- you're initializing this variable to a list and now setting it to a str?

Reply:
Changed all such functions' return type to string and made the title singular.